core/src/main/java/org/kohsuke/stapler/export/Property.java

```diff
@@ -187,98 +184,97 @@ private void writeValue(Type expected, Object value, TreePruner pruner, DataWrit
+                writer.endArray();
+                return;
+            }
+            if(value instanceof Map) {
+                if (verboseMap!=null) {// verbose form
+                    writer.startArray();
+                    for (Map.Entry e : ((Map<?,?>) value).entrySet()) {
+                        writeStartObjectNullType(writer);
+                        writer.name(verboseMap[0]);
+                        writeValue(null,e.getKey(),pruner,writer);
+                        writer.name(verboseMap[1]);
+                        writeValue(null,e.getValue(),pruner,writer);
+                        writer.endObject();
+                    }
+                    writer.endArray();
+                } else {// compact form
+                    writeStartObjectNullType(writer);
+                    for (Map.Entry e : ((Map<?,?>) value).entrySet()) {
+                        writer.name(e.getKey().toString());
+                        writeValue(null,e.getValue(),pruner,writer);
+                    }
+                    writer.endObject();
+                }
+                return;
+            }
+            if(value instanceof Date) {
+                writer.valuePrimitive(((Date) value).getTime());
+                return;
+            }
+            if(value instanceof Calendar) {
+                writer.valuePrimitive(((Calendar) value).getTimeInMillis());
+                return;
+            }
+            if(value instanceof Enum) {
+                writer.value(value.toString());
+                return;
             }
-            return;
-        }
-        if(value instanceof Date) {
-            writer.valuePrimitive(((Date) value).getTime());
-            return;
-        }
-        if(value instanceof Calendar) {
-            writer.valuePrimitive(((Calendar) value).getTimeInMillis());
-            return;
-        }
-        if(value instanceof Enum) {
-            writer.value(value.toString());
-            return;
         }
 
-        // otherwise handle it as a bean
         try {
             writer.type(expected, value.getClass());
         } catch (AbstractMethodError _) {
             // legacy impl that doesn't understand it
         }
-        writer.startObject();
-        Model model = null;
-        try {
-            model = owner.get(c, parent.type, name);
-        } catch (NotExportableException e) {
-            if (skipIfFail) {
-                Logger.getLogger(Property.class.getName()).log(Level.FINE, e.getMessage());
-            } else {
-                throw e;
-            }
-            // otherwise ignore this error by writing empty object
-        }
-        if(model!=null)
+
+
+        if(model!=null) {
+            writer.startObject();
             model.writeNestedObjectTo(value, pruner, writer);
-        writer.endObject();
+            writer.endObject();
+        }else if(skipIfFail){
+            writer.startObject();
+            writer.endObject();
+        }
```

Comment:
Either throw an exception here, or move this whole `if(skipIfFail)` up into the end of the `catch` block